### PR TITLE
ServiceBus: fixes for renew lock tracing

### PR DESCRIPTION
**Pull request by lmolkova**
Tracing for renew lock was not working correctly and traced the overall renewing operation. It should trace individual requests instead.

E.g. this shows auto-lock renewal.

image

this change fixes it and also improves accessing message context for tracing.

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/FluxAutoLockRenew.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/FluxAutoLockRenew.java:5]: Just add a quick entry in CHANGELOG with the fixes. :)

File: sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/TracingIntegrationTests.java [hunk 2]
```diff
@@ -150,6 +154,90 @@ public void sendAndReceive() throws InterruptedException {
...
+        ServiceBusMessage message = new ServiceBusMessage(CONTENTS_BYTES);
+        StepVerifier.create(sender.sendMessage(message)).verifyComplete();
+
+        CountDownLatch processedFound = new CountDownLatch(1);
+        spanProcessor.notifyIfCondition(processedFound, s -> s.getName().equals("ServiceBus.process"));
+
+        AtomicReference<ServiceBusReceivedMessage> received = new AtomicReference<>();
+        StepVerifier.create(receiver.receiveMessages()
+            .take(1)
+            .doOnNext(msg -> received.set(msg))
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/TracingIntegrationTests.java:168]: Same here with the doOnNext pattern

**Review by conniey — APPROVED** [review]
Just a couple of comments. Thanks!

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/instrumentation/ContextAccessor.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        return receiveAccessor.getContext(message);
+    }
+
+    public static Context getContext(ServiceBusMessage message) {
+        return sendAccessor.getContext(message);
+    }
+
+    public static void setReceiveMessageContextAccessor(final ReceiveMessageContextAccessor accessor) {
+        receiveAccessor = accessor;
+    }
```
Comment (anuchandy) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/instrumentation/ContextAccessor.java:37]: nit; missing a new line

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/instrumentation/ServiceBusSenderInstrumentation.java [hunk 4]
```diff
@@ -1,45 +1,43 @@
...
+import com.azure.messaging.servicebus.ServiceBusMessage;
 import reactor.core.publisher.Mono;
 
 import java.util.List;
 
 /**
  * Contains convenience methods to instrument specific sender calls with traces and metrics.
  */
-class ServiceBusSenderInstrumentation {
+public class ServiceBusSenderInstrumentation {
```
Comment (anuchandy) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/instrumentation/ServiceBusSenderInstrumentation.java:17]: nit; can this be `final`?

**Review by anuchandy — APPROVED** [review]
thanks

File: sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/TracingIntegrationTests.java [hunk 5]
```diff
@@ -150,6 +154,90 @@ public void sendAndReceive() throws InterruptedException {
...
+
+        AtomicReference<ServiceBusReceivedMessage> received = new AtomicReference<>();
+        StepVerifier.create(
+            sessionReceiver
+                .acceptSession(sessionId)
+                .flatMapMany(rec -> rec
+                    .renewSessionLock()
+                    .thenMany(rec.receiveMessages()
+                        .take(1)
+                        .doOnNext(msg -> {
```
[the remark this answers is not in the document]
Comment (lmolkova) [reply sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/TracingIntegrationTests.java:218]: thank you for optimizing my naive reactor code :)

**Pull request merged**